agent/util/subtree_unix_test.go

```diff
@@ -72,58 +72,62 @@ func TestGetPIDsToKill(t *testing.T) {
 }
 
 func TestKillSpawnedProcs(t *testing.T) {
-	ctx := context.Background()
-	for testName, test := range map[string]func(*testing.T){
-		"expired context": func(t *testing.T) {
+	for testName, test := range map[string]func(ctx context.Context, t *testing.T){
+		"ErrorsWithContextTimeout": func(ctx context.Context, t *testing.T) {
 			expiredContext, cancel := context.WithTimeout(ctx, -time.Second)
 			defer cancel()
 
 			err := KillSpawnedProcs(expiredContext, "", "", grip.GetDefaultJournaler())
 			assert.Error(t, err)
 			assert.Equal(t, ErrPSTimeout, errors.Cause(err))
 		},
-		"cancelled context": func(t *testing.T) {
+		"ErrorsWithContextCancelled": func(ctx context.Context, t *testing.T) {
 			cancelledContext, cancel := context.WithCancel(ctx)
 			cancel()
 
 			err := KillSpawnedProcs(cancelledContext, "", "", grip.GetDefaultJournaler())
 			assert.Error(t, err)
 			assert.NotEqual(t, ErrPSTimeout, errors.Cause(err))
 		},
-		"not cancelled": func(t *testing.T) {
+		"SucceedsWithNoContextError": func(ctx context.Context, t *testing.T) {
 			err := KillSpawnedProcs(ctx, "", "", grip.GetDefaultJournaler())
 			assert.NoError(t, err)
 		},
 	} {
-		t.Run(testName, test)
+		t.Run(testName, func(t *testing.T) {
+			ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+			defer cancel()
+
+			test(ctx, t)
+		})
 	}
 
 }
 
 func TestWaitForExit(t *testing.T) {
-	ctx, cancel := context.WithCancel(context.Background())
-	defer cancel()
-
-	for testName, test := range map[string]func(*testing.T){
-		"non-existent process": func(t *testing.T) {
-			waitCtx, cancel := context.WithTimeout(ctx, 10*time.Second)
-			defer cancel()
-			pids, err := waitForExit(waitCtx, []int{1234567890})
+	for testName, test := range map[string]func(ctx context.Context, t *testing.T){
+		"DoesNotReturnNonexistentProcess": func(ctx context.Context, t *testing.T) {
+			pids, err := waitForExit(ctx, []int{1234567890})
+			assert.NoError(t, ctx.Err())
```

Comment:
if the context is cancelled, then this will still be flaky right? do we want it to be like, if there's no context error, then check the below conditions? (Or do we expect this to now only happen consistently?)